lib/encode_url.js

```diff
@@ -4,11 +4,21 @@ const { parse, format } = require('url');
 
 function encodeURL(str) {
   if (parse(str).protocol) {
-    return format({
+    const obj = Object.assign({}, {
       protocol: parse(str).protocol,
       hostname: parse(str).hostname,
```

Comment:
Should it use `host` instead of `hostname`? `host` includes port while `hostname` doesn't.

And `auth` (`user:pass`) might be needed in output result (without encoded) as well.

Reply:
Fixed